### DISCUSSION TITLE
TMA2 서버 구축중 socket.io 의 502 에러.

### DISCUSSION THREAD

**Discussion by jshs20211402** [luftaquila/monolith#28] [Q&A] [answered]
안녕하세요 GIST의 GBungE팀에서 로깅 개발을 맡은 학생입니다.

구글 클라우드로 서버를 만들어보고 있었습니다. 사용한 OS 버전은 ubuntu 20.04.6 LTS입니다.

TMA2 서버 구축 문서에 따라 서버를 만들고, 구글 애널리틱스와 카카오맵 api도 받아와서 띄우는 데 성공했습니다.

다만 socket.io가 계속 502 bad gateway 오류가 떠 도움을 요청하고자 합니다.

monolith.conf입니다.

```
location /socket.io {
    rewrite ^/socket.io/(.*) /socket.io/$1 break;
    proxy_pass http://localhost:8080/;
  }
```

다른 포트로 해보았으나, 되지 않았습니다..

혹시나 하고 방화벽 설정에서 모든 ip의 8080포트를 입,출력 모두 열어놓았는데도 되지 않습니다..

![image](https://github.com/user-attachments/assets/3b5143f7-d0ac-4d04-919f-63af1dca2e7a)

문제에 관해 아시는 바가 있다면 조언을 받을 수 있을까요?

**Comment by luftaquila** [comment]
안녕하세요.

`proxy_pass` 는 외부 포트로 들어온 트래픽을 내부에서 리버스 프록시로 8080번 포트로 포워딩하는 구문입니다. 외부에는 80번과 443번만 노출되면 됩니다.

일단 올려주신 설정에 80번과 443번 outbound 허용 규칙이 없는 것 같네요. 해당 포트에 egress rule을 추가한 다음 다시 시도해 보시고, 잘 된다면 8080번 rule은 삭제하시면 됩니다.

**Reply by luftaquila** [reply, marked as answer]
추가로 502가 뜬다는 건 nginx가 요청을 받아보고는 있다는 것 같은데요, 엔진엑스 로그 디렉터리 (일반적으로 `/var/log/nginx` 입니다) 에 error.log 파일이 있습니다. 해당 파일에서 문제 원인을 찾으실 수 있을 듯 합니다. `tail -f /var/log/nginx/error.log` 로 실시간으로 확인하실 수 있습니다.

그리고 서버 node.js 프로그램을 8080번 포트에서 실행하고 있는게 맞는지도 확인해 보세요.

**Reply by jshs20211402** [reply]
감사합니다. 덕분에 이유는 알 수 없지만 localhost에 접속이 불가? 한거 같은 오류를 알게 되었습니다.

구글 클라우드 플랫폼에서 NGINX가 기본으로 빌드 되어있는 이미지를 가져오면서 런타임 구성기? 같은 것을 설치하니 정상적으로 작동하는 것을 확인하였습니다.

Ubuntu 환경은 아니지만 기반이 되는 Debian 에서 실행중입니다.

**Reply by luftaquila** [reply]
Google Cloud는 사용해 본 적이 없어서 잘 모르겠네요. 아무튼 해결되셨다니 다행입니다.
